internal/retry/policy.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package retry
+
+import (
+	"context"
+	"time"
+
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/internal/backoff"
+)
+
+// PolicyProvider returns a retry policy to use for the request,
+// if no policy is available it should return a default policy.
+type PolicyProvider func(context.Context, *transport.Request) *Policy
+
+var defaultPolicy = Policy{
+	retries: 0,
+	timeout: time.Second,
+	backoff: backoff.FixedBackoff(time.Millisecond * 200).Backoff,
+}
+
+// Policy defines how a retry will be applied.  It contains all the information
+// needed to preform a retry.
+type Policy struct {
+	// retries is the number of attempts we will retry (after the
+	// initial attempt.
+	retries uint
+
+	// timeout is the Timeout we will enforce per request (if this
+	// is less than the context deadline, we'll use that instead).
```

Comment:
Should `less` be `more` here? And would naming this `maxRequestTimeout` be an accurate name?